flixel/util/FlxDirections.hx

```diff
@@ -0,0 +1,117 @@
+package flixel.util;
+
+/**
+ * Uses bit flags to create a list of orthogonal directions. useful for
+ * many `FlxObject` features like `allowCollisions` and `touching`.
+ * @since 4.10.0
+ */
+@:enum abstract FlxDirections(Int) from Int to Int
+{
+	/** Generic value for "left". */
+	var LEFT    = 0x0001;
+	/** Generic value for "right". */
+	var RIGHT   = 0x0010;
+	/** Generic value for "up". */
+	var UP      = 0x0100;
+	/** Generic value for "down". */
+	var DOWN    = 0x1000;
+	/** Special-case constant meaning no directions. */
+	var NONE    = 0x0000;
+	
+	/**
+	 * Directions values can't be constructed from other direction values, due to a bug in haxe.
+	 * https://github.com/HaxeFoundation/haxe/issues/10237
+	 * We'll just define all these using literal hex values, otherwise they can't be used as
+	 * default values for function arguments.
+	 */
+	
+	/** Special-case constant meaning "up". */
+	var CEILING = 0x0100;// UP;
+	/** Special-case constant meaning "down" */
+	var FLOOR   = 0x1000;// DOWN;
+	/** Special-case constant meaning "left" and "right". */
+	var WALL    = 0x0011;// LEFT | RIGHT;
+	/** Special-case constant meaning any, or all directions. */
+	var ANY     = 0x1111;// LEFT | RIGHT | UP | DOWN;
```

Comment:
I would've just inlined these into the doc comments, for instance:

```
Special-case constant meaning `UP`.
```

or

```
Special-case constant meaning any, or all directions (`LEFT | RIGHT | UP | DOWN`).
```

-----

Looking at this more closely though, I think we should probably just get rid of `CEILING`, `FLOOR` and `WALL`, and only keep `ANY`. The first three only really made sense for collisions, and this is meant to be a generic abstract for directions. After all, the old convenience constants still exist on `FlxObject` anyway.

Reply:
I'm not a fan of removing helper values like ceiling or wall, if anything I'd rather add more case specific values like `X_AXIS` and `Y_AXIS`, i find them indicative of their broad uses. `WALLS` in particular was used in the snippet that helped me first understand direction flags in flixel. the `FlxObject` constants have little value to me, also. I would remove them completely if I didn't care about backwards compatibility.